### ISSUE TITLE
StoreState Function no longer working after updating to 2.3.0

### ISSUE THREAD

**Issue by ninja-warrior** [ioBroker/ioBroker.influxdb#196]
Hey guys,

since I have updated my iobroker.influxdb adapter to latest version I am no longer able to write state changes via JavaScript to my influxdb database. 

- Adapter Version: 2.3.0
- JS Controller Version: 3.3.21

Here is my code snippet which I use for writing state changes to influxdb database:

```
function storeDB(id, ts, val, source){ 
    sendTo('influxdb.0', 'storeState', {
        id: id,
        state: {ts: ts,
                val: val,
                ack: true,
                from: source
                }
    });
}
```

I get the following error message, when using the function in my scripts:

`influxdb.0 (159483) States system pmessage io.messagebox.system.adapter.influxdb.0 {"command":"storeState","message":{"id":"javascript.0.EnergyStatistics.PV_forecast_tomorrow","state":{"ts":1639667943605,"val":0,"ack":true,"from":"script.js.common.EnergyStatistic.PVForecast"}},"from":"system.adapter.javascript.0","_id":56204994} Cannot read property 'influxdb.0' of undefined`

Before I did the update anything worked fine. 

Any ideas what I need to change?

Thanks and best regards,
David

**Comment by Apollon77** [comment]
This error is caused by you writing a value into the database which is not configured to be historized.

Why you have at all such a script and not use normal way?

**Comment by ninja-warrior** [comment]
Thanks for your reply. I've got the following use case:

I own a photovoltaic system with kostal plenticore inverter. I have integrated this inverter to ioBroker using the plenticore adapter. The plenticore adapter also provides power forecast for each hour of today and tomorrow. For this reason the plenticore adapter manages one state for  every hour within the forecast intervall. 

I would like to validate the accuracy of the forecast made by plenticore adapter by comparing the actual values against the forecasts. For this reason I want to show these values in a timeseries using grafana.

The challenge is that grafana timeseries only works with historized data. So for that reason I need to write the single state values for each hour to a common influxdb table.

This worked fine for weeks. Why do we need to restrict the storeState functionality in Javascript so hard?

**Comment by Apollon77** [comment]
lets say it that way ... it was always intended to be that way because all configurations and all management functions are only available when we have a config for this datapoint. I will "fix" it in 2.4.0 so that it works again without killing this "unintended behaviour", but it can break anytime again ...

WIll be in 2.4.0

**Comment by ninja-warrior** [comment]
Thanks a lot!
